great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -309,6 +313,39 @@ def run(
 
         return expectation_suite
 
+    def reconcile_batch_requests_in_builders(
+        self, batch_request: Union[dict, BatchRequest, RuntimeBatchRequest]
+    ) -> None:
+        """
+        Profiler "batch_request" reconciliation involves combining existing Profiler state, instantiated from Profiler configuration
+        (e.g., stored in a YAML file managed by the Profiler store), with the batch request overrides, provided at run time.
+
+        The provided batch request is propagated to the following relevant Builders attributes (as applicable):
+            - ParameterBuilders
+            - ColumnDomainBuilder
+              - We default to the latest value as a sensible default
+
+        The reconciliation logic for "batch_request" is of the "replace" nature: the provided data is consistently applied, regardless
+        of existing Builder state.
+
+        Args:
+            batch_request: Data provided at runtime used to hydrate nested builder attributes
+        """
+        if isinstance(batch_request, dict):
+            batch_request = get_batch_request_from_acceptable_arguments(**batch_request)
+
+        # TODO(cdkini): Let's use setters
+        for rule in self.rules:
+            domain_builder = rule.domain_builder
+            if isinstance(domain_builder, ColumnDomainBuilder):
+                domain_builder._batch_request = copy.deepcopy(batch_request)
+                domain_builder._batch_request.data_connector_query = {"index": -1}
+
+            parameter_builders = rule.parameter_builders
+            if parameter_builders:
+                for parameter_builder in parameter_builders:
+                    parameter_builder._batch_request = copy.deepcopy(batch_request)
```

Comment:
This is all WIP but looking to confirm that this general idea is appropriate. It's not really "reconciliation" since we're blindly applying it to builders but I thought I'd keep the naming convention consistent.